### PR TITLE
Add scaffolding for DocumentQuery execution, #241.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 1]
```diff
@@ -379,29 +368,53 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
...
 
         feedOptions.setEnableCrossPartitionQuery(isCrossPartition);
 
         return getDocumentClient().queryDocuments(selfLink, sqlQuerySpec, feedOptions);
     }
 
-    private FeedResponse<Document> executeQuery(@NonNull com.microsoft.azure.documentdb.SqlQuerySpec sqlQuerySpec,
-                                                FeedOptions feedOptions, String collectionName) {
-        final String selfLink = toCollectionSelfLink(collectionName);
+    private com.microsoft.azure.documentdb.FeedResponse<Document> executeQuery(
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Only add two basic executeQuery method for Async. we can add more if need base on this.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 2]
```diff
@@ -379,29 +368,53 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
...
+    public Observable<com.microsoft.azure.cosmosdb.Document> executeQueryAsyncDocument(
+            @NonNull SqlQuerySpec sqlQuerySpec, @NonNull String collectionName, @NonNull FeedOptions options) {
+        return executeQueryAsync(sqlQuerySpec, collectionName, options)
+                .subscribeOn(Schedulers.io())
+                .map(FeedResponse::getResults)
+                .subscribeOn(Schedulers.io())
+                .flatMap(Observable::from);
+    }
+
+    // Use public for now, will change to private after referenced
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Only add two basic executeQuery method for Async. we can add more if need base on this.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformation.java [hunk 3]
```diff
@@ -91,7 +86,7 @@ public String getPartitionKeyFieldValue(T entity) {
         return partitionKeyField == null ? null : (String) ReflectionUtils.getField(partitionKeyField, entity);
     }
 
-    public boolean isIdFieldAsPartitonKey() {
+    public boolean isIdFieldAsPartitionKey() {
         final String keyName = getPartitionKeyFieldName();
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformation.java]: This `isIdFieldAsPartitionKey` will not be used in this class?

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/DocumentDbEntityInformation.java]: yes, this is for corner case, partitionkey act on Id Field.

**Pull request merged**